### PR TITLE
refactor YGE code (done) & start on bulk payouts (done)

### PR DESCRIPTION
**Pull request by owocki**
##### Description

refactor YGE code (move to jquery, deprecate most of the rote JS, rename to 'onepager' which is more represntative of the template) and start on bulk payouts

TODOS:

* [x] Destroy YGE JS dependancies
* [x] Remove ETHos/coin redeem apps
* [x] Rename YGE to something that represents Gitcoin's app ( 'onepage' maybe )
* [x] rebuild Tips send flow
* [x] rebuild Tips receive flow
* [x] support for tokens
* [x] provide legacy way for ppl to redeem their legacy tips
* [X] Build Bulk Payouts
* [x] Analytics Updates (to disinclude pending tips)

TODO down the line after we are satisfied that all tips have been claimed -- delete `receive_legacy.html`, receive url route, and `onepager/js/quarantine`

## Demo of Bulk Payouts

screen recording 2018-07-11 at 09 07 pm

## Demo of NewTips Send
screen recording 2018-07-11 at 09 10 pm

## Demo of NewTips Receive

screen recording 2018-07-11 at 09 13 pm

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)

tips / youvegoteth

##### Refers/Fixes

https://github.com/gitcoinco/web/issues/1422

##### v2 ticket https://github.com/gitcoinco/web/issues/1720

### CODE REVIEW

**Comment by mbeacom** [comment]
screenshot 2018-07-09 14 42 21
🎸 `\m/`

**Comment by owocki** [comment]
@mbeacom @thelostone-mc @SaptakS 

i've reached feature complete on the tips refactor and it's now working on rinkeby for both tokens and for ETH.  mind taking a look?

the only WIP part left is the bulk payouts.. i can't do that today, but plan to turnaround code review and do it tomorrow on the next non-meeting-day.

**Comment by owocki** [comment]
aiming to merge this this week

**Comment by owocki** [comment]
v2 ticket https://github.com/gitcoinco/web/issues/1720

**Comment by owocki** [comment]
@mbeacom @SaptakS @thelostone-mc pls review..  @PixelantDesign pls QA

File: app/app/urls.py [hunk 1]
```diff
@@ -143,12 +151,6 @@
     url(r'^tools/?', dashboard.views.toolbox, name='tools'),
     url(r'^gas/?', dashboard.views.gas, name='gas'),
 
-    # redeem coin
```
Comment (owocki) [inline app/app/urls.py]: whats fun is that we could now programatically create tips and use those to redeem coin airdrops

Comment (owocki) [reply app/app/urls.py]: or even extend the current receive flow to accept 721s

File: app/app/urls.py [hunk 2]
```diff
@@ -120,11 +121,18 @@
...
-    url(r'^tip/?', dashboard.views.send_tip, name='tip'),
+    url(
+        r'^tip/receive/v2/(?P<pk>.*)/(?P<txid>.*)/(?P<network>.*)?',
+        dashboard.tip_views.receive_tip_v2,
+        name='receive_tip'
+    ),
+    url(r'^tip/receive/?', dashboard.tip_views.receive_tip_legacy, name='receive_tip_legacy'),
+    url(r'^tip/send/4/?', dashboard.tip_views.send_tip_4, name='send_tip_4'),
+    url(r'^tip/send/3/?', dashboard.tip_views.send_tip_3, name='send_tip_3'),
+    url(r'^tip/send/2/?', dashboard.tip_views.send_tip_2, name='send_tip_2'),
```
Comment (thelostone-mc) [inline app/app/urls.py]: @owocki @SaptakS  @mbeacom  should we rename this into something more meaning full ? ( we can do it in a follow up PR)
 `/2`, `/3` seems cryptic :P

**Comment by thelostone-mc** [comment]
Looks like we have two different versions of font awesome ! `assets/onepager/fonts/*`
Could we remove this ? 

@mbeacom this is probably why the icons don't load on those pages as the font-awesome version is outdated

**Comment by mbeacom** [comment]
Latest changes are on staging

**Comment by PixelantDesign** [comment]
Very cool! 

- Should we denote some where that Basic Payout is paying 1 person and Advanced Payout pays multiple People? It's not elegant, but more explicit. Perhaps in parenthesis small underneath?
- What do you think about renaming 'Refund' to 'Bounty Stake'? It technically is a refund, but I'm not sure if users really need to understand how all of that works. They just need to know this is X amount I'm starting with as I think about how to divide up this money.
- Should we have a disclaimer somewhere above the payout button that the funder will need to pay the separate gas prices for each transaction/person? I'm not sure if this is technically possible but, it would be awesome to be able to pay one gas price for all 4 people.
- Otherwise, mostly visual clean up which we can do when Saptak and Aditya are back.

**Comment by owocki** [comment]
some TODOS on my side:
- All done page after info is submitted 
- Also- make sure to add local storage info about bounty

**Comment by owocki** [comment]
> Any chance we can split it off into a separate django module? 🤔

yes we can!  feel free to add a ticket for it 

> Cookies keep popping up even after accepting (is it set in localStorage? )

i haven't changed any of this logic

> . Should we create a separate PR for the tip flow changes or handle the url restructuring here?

i'm not sure what value that would bring.  let me know what you think it does for us

> Should we denote some where that Basic Payout is paying 1 person and Advanced Payout pays multiple People? It's not elegant, but more explicit. Perhaps in parenthesis small underneath?

Yes..  Where should this go?  Right now theres already a "switch to basic payout" or "switch to advanced payout" link under the header.

> What do you think about renaming 'Refund' to 'Bounty Stake'? It technically is a refund, but I'm not sure if users really need to understand how all of that works. They just need to know this is X amount I'm starting with as I think about how to divide up this money.

Sure, I will update this.

> Should we have a disclaimer somewhere above the payout button that the funder will need to pay the separate gas prices for each transaction/person? I'm not sure if this is technically possible but, it would be awesome to be able to pay one gas price for all 4 people.

Itd be possible in a v2.  At least, for all tip transactions we could do it.  Can't do it for the 'cancel tx' though.

> Otherwise, mostly visual clean up which we can do when Saptak and Aditya are back.

Great

**Comment by owocki** [comment]
@mbeacom any way to disable sticker-ci for this PR?

File: app/app/urls.py (hunk 2 above)
Comment (owocki) [reply app/app/urls.py]: the numbers are the step numbers.. sequential numbers seem straightforward to me..

**Comment by owocki** [comment]
@thelostone-mc 00d9b5e64c3474be784ea95cd50d37193ca85ab3 should satisfy your QA

**Pull request merged**